pkgs/tools/filesystems/ceph/default.nix

```diff
@@ -37,17 +41,22 @@
 , libcap_ng
 , libnl
 , libxml2
+, lmdb
 , lttng-ust
 , lua
+, lvm2
 , lz4
 , oath-toolkit
 , openldap
-, python311
+, parted
+, python311 # to get an idea which Python versions are supported by Ceph, see upstream `do_cmake.sh` (see `PYBUILD=` variable)
 , rdkafka
 , rocksdb
 , snappy
+, openssh
 , sqlite
 , utf8proc
+, xfsprogs
```

Comment:
Non-blocking and just a ~~small~~ note: given that Filestore is [declared unsupported in Reef onwards](https://docs.ceph.com/en/squid/rados/configuration/filestore-config-ref/), this has prompted me to draft up an issue to discuss what to do with Filestore in the future. I would gravitate towards adding a `withFilestore` flag to the Ceph build defaulting to true, that users can explicitly set to true so that at least in the future maintainers aren't faced with "should we drop this" if something breaks, rather than at least having a toggle for fully supporting a Bluestore build and supporting Filestore as a best-effort thing or at least providing a way to fail gracefully for those who opted in to that flag so that their version of Ceph doesn't go from supporting Filestore to not supporting it, rather than throwing a build failure or something, preventing from accidentally upgrading to a version that doesn't support their storage backend.

I'm all in favour for getting 19.2.0 merged for the 24.11 window, but if we could squeeze something like this in here it would probably be nice (since it could be announced in the release notes ahead of time, who knows *when* issues arise).
As said I have an issue drafted out for discussing this properly and moving it out of the context of the Squid release, and since any such change would likely be non-breaking there's still a large enough window for getting something prepared in 24.11 (even if it's just a reminder in the release note).

Reply:
I think the idea is fine but I'm not sure if it's worth it:

Given that FileStore is already deprecated now, I'd be surprised if somebody set `withFilestore = true` after we add it.

If you want to do it anyway, I'll support it.

Reply:
- #347485